pkg/uio/reader.go

```diff
@@ -46,3 +47,30 @@ func ReaderAtEqual(r1, r2 io.ReaderAt) bool {
 	}
 	return bytes.Equal(c, d) && reflect.DeepEqual(r1err, r2err)
 }
+
+// ReadIntoFile reads all from io.Reader into the file at given path.
+//
+// If the file at given path does not exist, a new file will be created.
+// If the file exists at the given path, but not empty, it will be truncated.
+func ReadIntoFile(r io.Reader, p string) error {
+	f, err := os.OpenFile(p, os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	b := make([]byte, 0, 512)
```

Comment:
is it not possible to use io.Copy here?
func Copy(dst Writer, src Reader) (written int64, err error)

Reply:
Done.

I was likely blind... when previous suggestion suggested CopyBuffer.  Read into impl of io.Copy, it is better. https://cs.opensource.google/go/go/+/refs/tags/go1.16.7:src/io/io.go;l=401.

I selected CopyBuffer instead, to opt in 512 bytes buffer as I tested it for my use case that. However, I don't think the default buffer of size `1024` in io.Copy would make it OOM again during downloading. Just avoided some time for a repeat manual test with a lot of steps.